textarea/elfinder-sakai/src/java/org/sakaiproject/elfinder/sakai/SakaiFsService.java

```diff
@@ -194,7 +200,14 @@ public String getVolumeId(FsVolume volume) {
 			if("!admin".equals(currentSiteId) || "~admin".equals(currentSiteId)){
 				continue;
 			}
-			volumes.add(getSiteVolume(currentSiteId));
+			try
+			{
+				volumes.add(getSiteVolume(siteService.getSiteVisit(currentSiteId).getId()));
+			} catch (IdUnusedException e) {
+				log.warn("Unexpected IdUnusedException in " + e.getClass().getName() + ": " + e.getMessage());
```

Comment:
With SLF4J, the better format of doing log statements is as follows:

```
log.warn("Unexpected IdUnusedException in {}: {}", e.getClass().getName(), e.getMessage());
```

Using this new style avoids unnecessary String concatenation (which can be expensive) when the logger is set to anything other than WARN (or whatever level your logger statement is sending).

Reply:
Also I think the class name may be unnecessary here; pretty sure the logger statement is probably going to spit out this information for you automatically.